src/cat.ts

```diff
@@ -0,0 +1,319 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { minimize_Powell } from 'optimization-js';
+import { Stimulus, Zeta } from './type';
+import { itemResponseFunction, fisherInformation, normal, findClosest } from './utils';
+import { validateZetaParams, fillZetaDefaults } from './corpus';
+import seedrandom from 'seedrandom';
+import _clamp from 'lodash/clamp';
+import _cloneDeep from 'lodash/cloneDeep';
+
+const abilityPrior = normal();
+
+export interface CatInput {
```

Comment:
CatInput is just a configuration interface that defines the setup options for creating a Cat, like method or theta. On the other hand, Cat is the actual class that manages the behavior and state.

Basically, CatInput is for setup, and Cat does the work!

Reply:
i am confused. does it mean the user need to create CatInput instead of Cat?

if yes, can you add documentation in the README to explain how to use jsClowder 

And will that conflict the use if user only wants to use jsCAT instead of jsClowder?

Reply:
We clarified this in person I think but just to follow up here. `CatInput` is just an interface defining the input format that `Cat` expects when it is being instantiated. The user creates a new `Cat` instance by passing in parameters that conform to the `CatInput` interface. It's actually already defined in jsCat. Emily didn't add it in this PR. @AnyaWMa , you added it two years ago in [this commit](https://github.com/yeatmanlab/jsCAT/commit/60571e69318c20bbf575a80ba02cefd6b17af55e).

Reply:
I think we clarified this in our Slack huddle but I'll just document the conversation here. `CatInput` is an interface defining the input format that `Cat` expects when it is being instantiated. The user creates a new `Cat` instance by passing in parameters that conform to the `CatInput` interface. It's actually already defined in jsCat. Emily didn't add it in this PR. @AnyaWMa , you added it two years ago in [this commit](https://github.com/yeatmanlab/jsCAT/commit/60571e69318c20bbf575a80ba02cefd6b17af55e).